### PR TITLE
Misc updates

### PR DESCRIPTION
**Pull request by adamshire123**
### What does this PR do?

- Updates Readme instructions for creating sample data and local testing
- Adds a check for valid SAP sequence number, and a test for the check.
- bumps dependency versions

### Helpful background context
During testing we learned that SAP sequence numbers must be at least three digits long. 
The production sequence number is in the thousands and shouldn't reset, so this shouldn't be an issue
in production, but it was during testing where our fake sap sequence number was '1'. 

The process to create sample data changed slightly. `README.me` was updated accordingly.

### Includes new or updated dependencies?

YES

### Developer

- [x] All new ENV is documented in README (or there is none)
- [x] Stakeholder approval has been confirmed (or is not needed)

### Code Reviewer

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [ ] There are appropriate tests covering any new functionality
- [ ] The documentation has been updated or is unnecessary
- [ ] The changes have been verified
- [ ] New dependencies are appropriate or there were no changes

### CODE REVIEW

File: sapinvoices/sap.py [hunk 1]
```diff
@@ -573,11 +577,16 @@ def generate_next_sap_sequence_number() -> str:
...
+
+    sequence number must be at least three digits (one or two digit numbers should be
+    stored as zero-padded to three places)
     """
     ssm = SSM()
     sap_config = load_config_values()
     sap_sequence_parameter = ssm.get_parameter_value(sap_config["SAP_SEQUENCE_NUM"])
     split_parameter = sap_sequence_parameter.split(",")
+    if len(split_parameter[0]) < 3:
+        raise SapSequenceError("SAP sequence number must be three or more digits.")
```
Comment (ehanson8) [inline sapinvoices/sap.py]: I would include the value here, e.g. `raise SapSequenceError("Invalid SAP sequence:  '%s', number must be three or more digits.", split_parameter[0])`

File: tests/test_sap.py [hunk 2]
```diff
@@ -571,6 +571,16 @@ def test_generate_next_sap_sequence_number():
     assert new_sap_sequence == "1002"
 
 
+@pytest.mark.usefixtures("mocked_ssm_bad_sequence_number")
+def test_generate_next_sap_sequence_number_fail():
```
Comment (ehanson8) [inline tests/test_sap.py]: This works without the pytest decorator and just `def test_generate_next_sap_sequence_number_fail(mocked_ssm_bad_sequence_number):`. That appears to override the autouse `mocked_ssm` fixture

File: tests/test_sap.py [hunk 3]
```diff
@@ -571,6 +571,16 @@ def test_generate_next_sap_sequence_number():
...
+    ssm = SSM()
```
Comment (ehanson8) [inline tests/test_sap.py]: `SSM()` could be a fixture since it's called in several tests. I did something similar in the original `alma-scripts`

**Review by ehanson8 — COMMENTED** [review]
A few suggestions

File: tests/test_sap.py (hunk 2 above)
Comment (adamshire123) [reply tests/test_sap.py]: `@pytest.mark.usefixtures` makes it so that pylint doesn't complain about unused arguments.

Comment (ehanson8) [reply tests/test_sap.py]: Got it, that seems fine

File: sapinvoices/sap.py [hunk 4]
```diff
@@ -586,7 +586,10 @@ def generate_next_sap_sequence_number() -> str:
     sap_sequence_parameter = ssm.get_parameter_value(sap_config["SAP_SEQUENCE_NUM"])
     split_parameter = sap_sequence_parameter.split(",")
     if len(split_parameter[0]) < 3:
-        raise SapSequenceError("SAP sequence number must be three or more digits.")
+        raise SapSequenceError(
+            f"Invalid SAP sequence: '{split_parameter[0]}', number must be three or more"
+            "digits."
```
Comment (ehanson8) [inline sapinvoices/sap.py]: Add space before "digits"

**Review by ehanson8 — APPROVED** [review]
There's a `llama` reference in `README.md` that should be removed but otherwise looks great!

**Pull request merged**